integration_tests/tests/billing-and-transaction-view-tests.ts

```diff
@@ -0,0 +1,176 @@
+import { resourcePrefix } from "../helpers/envHelper";
+
+import {
+  deleteS3Events,
+  generatePublishAndValidateEvents,
+  TableNames,
+  TimeStamps,
+} from "../helpers/commonHelpers";
+import {
+  deleteDirectoryRecursiveInS3,
+  putObjectToS3,
+  checkIfS3ObjectExists,
+  S3Object,
+} from "../helpers/s3Helper";
+import path from "path";
+import fs from "fs";
+import { EventName, ClientId } from "../payloads/snsEventPayload";
+import { queryResponseFilterByVendorServiceNames } from "../helpers/queryHelper";
+
+const prefix = resourcePrefix();
+const bucketName = `${prefix}-storage`;
+
+describe("\nUpload invoice to standardised folder and verify billing and transaction_curated view query results matches with expected data \n", () => {
+  const folderPrefix = "btm_billing_standardised";
+  const testObject: S3Object = {
+    bucket: `${prefix}-storage`,
+    key: `${folderPrefix}/receipt.txt`,
+  };
+  beforeAll(async () => {
+    await deleteDirectoryRecursiveInS3(bucketName, "btm_transactions");
+    // uploading file to s3 will be removed once BTM-276 implemented
+    const file = "../payloads/receipt.txt";
+    const filePath = path.join(__dirname, file);
+    const fileStream = fs.createReadStream(filePath);
+    await putObjectToS3(testObject, fileStream);
+    const checkFileExists = await checkIfS3ObjectExists(testObject);
+    expect(checkFileExists).toBeTruthy();
+  });
+
+  test.each`
+    eventName                          | clientId     | eventTime                  | numberOfTestEvents | priceDiff   | qtyDiff | priceDifferencePercent | qtyDifferencePercent | billingPrice | billingQty | transactionPrice | transactionQty
+    ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client1"} | ${TimeStamps.CURRENT_TIME} | ${"2"}             | ${"0.0000"} | ${"0"}  | ${"0.0000"}            | ${"0"}               | ${"6.6600"}  | ${"2"}     | ${"6.6600"}      | ${"2"}
+    ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client1"} | ${TimeStamps.CURRENT_TIME} | ${"1"}             | ${"3.3300"} | ${"1"}  | ${"100.0000"}          | ${"100"}             | ${"6.6600"}  | ${"2"}     | ${"3.3300"}      | ${"1"}
+    ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client1"} | ${TimeStamps.CURRENT_TIME} | ${undefined}       | ${"6.6600"} | ${"2"}  | ${undefined}           | ${undefined}         | ${"6.6600"}  | ${"2"}     | ${undefined}     | ${undefined}
+  `(
```

Comment:
There should be an example where billingQty is less than transactionQty, and another where the billingPrice and billingQty are undefined I think.

Reply:
added test case of billing qty less than transaction qty